byway/src/main/webapp/destinations.js

```diff
@@ -0,0 +1,173 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+/*
+* Sets Progress Bar to correct location based on the page number
+* @param {int} pageNumber
+*/
+window.onload = function() {
+  setProgressBar(1);
+  getStartDestination();
+  getLocations();
+};
+
+const defaultCenter = {
+  lat: 40.712776,
+  lng:-74.005974
+};
+Object.freeze(defaultCenter);
+let userlatlng = {lat:null , lng: null};
+
+/*
+* Creates map and search boxes with autocomplete
+*/
+function initAutocomplete() {   
+  const map = new google.maps.Map(document.getElementById("map"), {
+    center: { lat: defaultCenter.lat, lng: defaultCenter.lng },
+    zoom: 13,
+    mapTypeId: "roadmap"
+  });
+
+
+  if (navigator.geolocation) {
+    navigator.geolocation.getCurrentPosition(function(position) {
+      userlatlng = {
+        lat: position.coords.latitude,
+        lng: position.coords.longitude
+      };
+      map.setCenter(userlatlng);
+    }, function() {
+         map.setCenter(defaultCenter);
+        });
+  } 
+  else {
+    window.alert("Geolocation failed");
```

Comment:
This is fine for now, but in a future pull request, you should think about ways you may want to troubleshoot with the user. E.g. give them to the option to try again, or reload the page, or go back to the start page, etc.

Reply:
okay!